File: scalechain-storage/src/test/scala/io/scalechain/blockchain/storage/index/RocksDatabasePerformanceSpec.scala
package io.scalechain.blockchain.storage.index

import java.io.File
import java.math.BigInteger

import io.scalechain.blockchain.proto.{Transaction, Hash}
import io.scalechain.blockchain.storage.{TransactionPoolIndex, Storage}
import io.scalechain.crypto.HashFunctions
import io.scalechain.util.GlobalStopWatch
import org.apache.commons.io.FileUtils
import org.scalatest._

import scala.collection.mutable.ListBuffer
import scala.util.Random

class RocksDatabasePerformanceSpec extends FlatSpec with KeyValueDatabasePerformanceTrait with ShouldMatchers with BeforeAndAfterEach {
  this: Suite =>

  Storage.initialize()

  var db : KeyValueDatabase = null

  override def beforeEach() {

    val testPath = new File("./target/unittests-RocksDatabasePerformanceSpec")
    FileUtils.deleteDirectory( testPath )
    db = new RocksDatabase( testPath )

    super.beforeEach()
  }

  override def afterEach() {
    super.afterEach()

    db.close()
  }
}

File: scalechain-storage/src/test/kotlin/io/scalechain/blockchain/storage/index/RocksDatabasePerformanceSpec.kt
package io.scalechain.blockchain.storage.index

import io.kotlintest.KTestJUnitRunner
import io.kotlintest.matchers.Matchers
import io.kotlintest.specs.FlatSpec
import java.io.File
import io.scalechain.blockchain.storage.Storage
import io.scalechain.test.BeforeAfterEach
import org.apache.commons.io.FileUtils
import org.junit.runner.RunWith

@RunWith(KTestJUnitRunner::class)
class RocksDatabasePerformanceSpec : FlatSpec(), Matchers, KeyValueDatabasePerformanceTrait {

  val testPath = File("./target/unittests-RocksDatabasePerformanceSpec")

  lateinit override var db : KeyValueDatabase

  override fun beforeEach() {
    FileUtils.deleteDirectory( testPath )

    db = RocksDatabase( testPath )

    super.beforeEach()
  }

  override fun afterEach() {
    super.afterEach()

    db.close()
  }

  init {
    Storage.initialize()
    addTests()
  }
}
